core/trino-main/src/test/java/io/trino/operator/exchange/TestUniformPartitionRebalancer.java

```diff
@@ -107,6 +107,120 @@ public void testRebalanceWithWriterSkewness()
                         ImmutableList.of(1, 0));
     }
 
+    @Test
+    public void testRebalanceEffectionOnContext()
```

Comment:
This test looks pretty generic. Please add a test which verifies the use case. For instance, due to a particular partition rebalance, a few writers priorities are affected and they go down in the priority queue such that now they are not considered skewed in the next `while` iteration.